query/src/sessions/session_status.rs

```diff
@@ -30,6 +30,7 @@ pub struct MutableStatus {
     abort: AtomicBool,
     current_database: RwLock<String>,
     session_settings: RwLock<Settings>,
+    current_user: RwLock<Option<String>>,
```

Comment:
it will the UserInfo in the feature not only a user name?

Reply:
I'm taking UserInfo at first but reverted it back to a simple user name, in the current code base UserInfo is a fat object:

```
  pub struct UserInfo {
      pub name: String,
      pub hostname: String,
      pub password: Vec<u8>,
      pub auth_type: AuthType,
      pub privileges: UserPrivilege,
      pub quota: UserQuota,
  }
```

the fields inside UserInfo might get mutated on the fly (like quota, privileges, even password), or mutated by another query instance. a conservative usage of UserInfo might be always fetching `UserInfo` from the metasrv, and make it transactional with the user-related operations.

in the case about session privilege, we can construct a unique identifier for the privilege grants by user_name + client_address.